azure-ts-dynamicresource/cdnCustomDomain.ts

```diff
@@ -0,0 +1,244 @@
+        endpointName: inputs.endpointName,
+        customDomainHostName: inputs.customDomainHostName,
+        httpsEnabled: inputs.httpsEnabled,
+      },
+    };
+
+    return {
+      id: result.id!,
+      outs: outs
+    };
+  }
+
+  async read(id: string, props: any): Promise<pulumi.dynamic.ReadResult> {
+    const inputs = props.inputs as CustomDomainOptions;
+    const cdnClient = await this.getCDNManagementClient();
+    const result = await cdnClient.customDomains.get(inputs.resourceGroupName, inputs.profileName, inputs.endpointName, this.name);
+    return {
+      id: result.id,
+      props: { ...result, inputs: { ...inputs }}
+    };
+  }
+
+  async delete(id: string, props: any): Promise<void> {
+    const inputs = props.inputs as CustomDomainOptions;
+    const cdnClient = await this.getCDNManagementClient();
+    try {
+      const result = await cdnClient.customDomains.deleteMethod(inputs.resourceGroupName, inputs.profileName, inputs.endpointName, this.name);
+      if (result._response.status >= 400) {
+        throw new Error("Error response received while trying to delete the custom domain.");
+      }
+
+      if (!result.resourceState) {
+        return;
+      }
+      if (result.resourceState !== 'Deleting') {
+        throw new Error(`Provisioning state of the custom domain was expected to be 'Deleting', but was ${ result.resourceState }.`);
+      }
+
+      console.log("The request to delete was successful. However, it can take a minute or two to fully complete deletion.");
+    } catch (err) {
+      // Due to a bug in the Azure CDN SDK, deletion will result in a 404 even though it succeeded.
+      // This console.log statement will at least keep the user aware of the problem.
+      // Once the bug is fixed in the SDK, we should never hit this unless the custom domain to be deleted
+      // cannot be genuinely found.
+      // See: https://github.com/Azure/azure-sdk-for-js/issues/2842.
+      console.log(err);
+    }
+  }
+
+  /**
+   * The only thing that the update method really updates in the custom domain is the HTTPS enablement.
+   */
+  async update (id: string, currentOutputs: CustomDomainOutputs, newInputs: CustomDomainOptions): Promise<pulumi.dynamic.UpdateResult> {
+    const cdnClient = await this.getCDNManagementClient();
+    if (newInputs.httpsEnabled) {
+      await cdnClient.customDomains
+        .enableCustomHttps(
+          newInputs.resourceGroupName,
+          newInputs.profileName,
+          newInputs.endpointName,
+          this.name);
+      
+      currentOutputs.inputs.httpsEnabled = true;
+      return {
+        outs: currentOutputs
+      };
+    }
+
+    await cdnClient.customDomains
+      .disableCustomHttps(
+        newInputs.resourceGroupName,
+        newInputs.profileName,
+        newInputs.endpointName,
+        this.name);
+
+    currentOutputs.inputs.httpsEnabled = true;
+    return {
+      outs: currentOutputs
+    };
+  }
+}
```

Comment:
note: i actually skipped most of this as i don't feel qualified to review a provider.  perhaps w can get @pgavlin to take a look.  It's something i want to ramp up on, so i'll read this and ask questions.  But i'd like someone more informed to take a deeper look.